openmdao/utils/assert_utils.py

```diff
@@ -47,6 +47,35 @@ def assert_warning(category, msg):
         raise AssertionError("Did not see expected %s: %s" % (category.__name__, msg))
 
 
+@contextmanager
+def assert_no_warning(category, msg):
+    """
+    Context manager asserting that a warning is not issued.
+
+    Parameters
+    ----------
+    category : class
+        The class of the warning.
+    msg : str
+        The text of the warning.
+
+    Raises
+    ------
+    AssertionError
+        If the warning is raised.
+    """
+    with reset_warning_registry():
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+            yield
+
+    for warn in w:
+        if (issubclass(warn.category, category) and str(warn.message) == msg):
+            raise AssertionError("Found warning: ", msg)
+        else:
+            break
```

Comment:
By putting a break here, you're only checking the first warning.  The else block should be removed.